third_party/blink/renderer/core/loader/resource/font_resource.cc

```diff
@@ -99,16 +99,30 @@ void FontResource::StartLoadLimitTimersIfNecessary(
   DCHECK(!font_load_long_limit_.IsActive());
   load_limit_state_ = LoadLimitState::kUnderLimit;
 
-  font_load_short_limit_ = PostDelayedCancellableTask(
-      *task_runner, FROM_HERE,
-      WTF::BindOnce(&FontResource::FontLoadShortLimitCallback,
-                    WrapWeakPersistent(this)),
-      kFontLoadWaitShort);
-  font_load_long_limit_ = PostDelayedCancellableTask(
-      *task_runner, FROM_HERE,
-      WTF::BindOnce(&FontResource::FontLoadLongLimitCallback,
-                    WrapWeakPersistent(this)),
-      kFontLoadWaitLong);
+  if (recordreplay::IsRecordingOrReplaying("avoid-weak-pointers")) {
+    font_load_short_limit_ = PostDelayedCancellableTask(
+        *task_runner, FROM_HERE,
+        WTF::BindOnce(&FontResource::FontLoadShortLimitCallback,
+                      WrapPersistent(this)),
+        kFontLoadWaitShort);
+    font_load_long_limit_ = PostDelayedCancellableTask(
+        *task_runner, FROM_HERE,
+        WTF::BindOnce(&FontResource::FontLoadLongLimitCallback,
+                      WrapPersistent(this)),
+        kFontLoadWaitLong);
+  }
+  else {
+    font_load_short_limit_ = PostDelayedCancellableTask(
+        *task_runner, FROM_HERE,
+        WTF::BindOnce(&FontResource::FontLoadShortLimitCallback,
+                      WrapWeakPersistent(this)),
+        kFontLoadWaitShort);
+    font_load_long_limit_ = PostDelayedCancellableTask(
+        *task_runner, FROM_HERE,
+        WTF::BindOnce(&FontResource::FontLoadLongLimitCallback,
+                      WrapWeakPersistent(this)),
+        kFontLoadWaitLong);
+  }
```

Comment:
@bhackett1024, I looked into this:
* I'm not sure where the existing wrapper would be?
* I'm not sure how to generalize, since there is no unambiguous base class between `Persistent` and `WeakPersistent` 🤷 

This fails:
```cpp
auto self = recordreplay::IsRecordingOrReplaying("avoid-weak-pointers")
                ? WrapPersistent(this)
                : WrapWeakPersistent(this);
```

with this error:
```
../../third_party/blink/renderer/core/loader/resource/font_resource.cc:105:19: error: conditional expression is ambiguous; 'BasicPersistent<[...], internal::StrongPersistentPolicy>' can be converted to 'BasicPersistent<[...], internal::WeakPersistentPolicy>' and vice versa
                  ? WrapPersistent(this)
                  ^ ~~~~~~~~~~~~~~~~~~~~
```

Any suggestions would be welcome 😄

Reply:
If there is a common base class between the two sides of the ternary it should be possible to resolve the ambiguity with a static_cast on one or both of the operands.

Reply:
@bhackett1024,
> If there is a common base class between the two sides of the ternary it should be possible to resolve the ambiguity with a static_cast on one or both of the operands.

The base class `BasicPersistent` wants conflicting type arguments, according to the error message above.

Open to concrete suggestions though (since my template-fu is certainly not the best) 🤷

Reply:
I can fix the copy-and-paste with a local `#define` → `#undef` if that's acceptable?